astropy/modeling/mappings.py

```diff
@@ -182,3 +183,117 @@ def inverse(self):
         """
 
         return self
+
+
+class UnitsMapping(Model):
+    """
+    Coerce input units (or dimensionless input) into new units without
+    conversion.  Used by Model.coerce_units to support units on otherwise
+    dimensionless models such as Polynomial1D.
+
+    Parameters
+    ----------
+    mapping : tuple
+        A tuple of (input_unit, output_unit) pairs, one per input, matched to the
+        inputs by position.  The first element of the each pair is the unit that
+        the model will accept (specify `units.dimensionless_unscaled` to accept
+        dimensionless input).  The second element is the unit that the model will
+        return.
+    input_units_equivalencies : dict, optional
+        Default equivalencies to apply to input values.  If set, this should be a
+        dictionary where each key is a string that corresponds to one of the
+        model inputs.
+    name : str, optional
+        A human-friendly name associated with this model instance
+        (particularly useful for identifying the individual components of a
+        compound model).
+    meta : dict-like, optional
+        Free-form metadata to associate with this model.
+
+    Examples
+    --------
+
+    >>> from astropy.modeling.models import Polynomial1D, UnitsMapping
+    >>> from astropy import units
+    >>> poly = Polynomial1D(1, c0=1, c1=2)
+    >>> model = UnitsMapping(((units.m, None),)) | poly
+    >>> model = model | UnitsMapping(((None, units.s),))
+    >>> model(units.Quantity(10, units.m))  # doctest: +FLOAT_CMP
+    <Quantity 21. s>
+    >>> model(units.Quantity(1000, units.cm)) # doctest: +FLOAT_CMP
+    <Quantity 21. s>
+    """
```

Comment:
I think an _additional_ useful example would be what happens when the value is the same but units change, i.e., `model(units.Quantity(10, units.cm))`

Reply:
Good point, I'll add that line to the existing example

Reply:
(added)